library/src/main/java/com/otaliastudios/zoom/internal/MatrixManager.kt

```diff
@@ -0,0 +1,445 @@
+package com.otaliastudios.zoom.internal
+
+import android.animation.*
+import android.annotation.SuppressLint
+import android.graphics.Matrix
+import android.graphics.RectF
+import android.view.animation.AccelerateDecelerateInterpolator
+import com.otaliastudios.zoom.*
+import com.otaliastudios.zoom.ZoomApi.RealZoom
+import com.otaliastudios.zoom.ZoomApi.Zoom
+import com.otaliastudios.zoom.internal.movement.PanManager
+import com.otaliastudios.zoom.internal.movement.ZoomManager
+
+/**
+ * Applies changes to the matrix, holds the content and container sizes and
+ * transformed rects, dispatches updates.
+ *
+ * It also uses [Controller] because it can start (and end) animations.
+ */
+internal class MatrixManager(
+        private val zoomManager: ZoomManager,
+        private val panManager: PanManager,
+        private val controller: Controller,
+        private val callback: Callback,
+        private val engine: ZoomEngine
+) {
+
+    internal interface Callback {
```

Comment:
I personally don't really like these inner classes (or interfaces) and would rather use a separate file for this interface and give it a proper name, but I guess it's personal preference so I would leave it up to you.